### PR TITLE
External authentication plugins

### PR DESCRIPTION
**Pull request by enolfc**
# Summary 

Provides a way to extend authentication with pluggable python classes that will refresh the credentials as needed. 

----

**Related issue :** #160

### CODE REVIEW

File: cloud_info_provider/auth_refreshers/__init__.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+"""
+Authentication refreshers
+
+Provides a pluggable to dynamically refresh authentication credentials if
```
Comment (gwarf) [inline cloud_info_provider/auth_refreshers/__init__.py]:
```suggestion
Provides a pluggable way to dynamically refresh authentication credentials if
```

File: cloud_info_provider/auth_refreshers/__init__.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+needed. The actual refresh mechanism depends greatly on the actual credential
+type and the provider used.
+
+The `auth_refresh` method is invoked every time the provider needs to rescope
```
Comment (gwarf) [inline cloud_info_provider/auth_refreshers/__init__.py]:
```suggestion
The `refresh` method is invoked every time the provider needs to rescope
```

File: cloud_info_provider/auth_refreshers/oidc_refresh.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
+import requests
+
+from cloud_info_provider import auth_refreshers
+from cloud_info_provider import exceptions
+
+
+class OidcRefreshToken(auth_refreshers.BaseRefresher):
+    def __init__(self, opts):
```
Comment (gwarf) [inline cloud_info_provider/auth_refreshers/oidc_refresh.py]: Is it required as it's the exact same code as the parent?

File: setup.cfg [hunk 4]
```diff
@@ -50,6 +50,8 @@ cip.collectors =
     clouddomain = cloud_info_provider.collectors.cloud:CloudCollector
     storage = cloud_info_provider.collectors.storage:StorageCollector
     compute = cloud_info_provider.collectors.compute:ComputeCollector
+cip.auth_refreshers=
```
Comment (gwarf) [inline setup.cfg]: Indentation looks weird.

Comment (enolfc) [reply setup.cfg]: Missing space corrected

**Pull request merged**